### PR TITLE
Remove ProjectController#toggle_watch method

### PR DESCRIPTION
**Pull request by saraycp**
We are replacing the old watchlist code with the new one step by step.

In this case, we focus on removing `ProjectController#toggle_watch` which is now replaced by `Webui::WatchedItemsController#toggle_watched_item`.

### CODE REVIEW

**Comment by dmarcoux** [comment]
I believe this PR can be rebased now that #13444 has been merged.

**Review by danidoni — APPROVED** [review]

File: src/api/config/routes/webui_routes.rb [hunk 1]
```diff
@@ -227,7 +227,6 @@
         "/project/monitor/#{request.query_parameters[:project]}#{url_string}"
       }, constraints: ->(request) { request.query_parameters['project'].present? }
       # TODO: this should be POST (and the link AJAX)
```
Comment (dmarcoux) [inline src/api/config/routes/webui_routes.rb:229]: Is this comment still relevant?

**Comment by saraycp** [comment]
`codeclimate/total-coverage` and `codeclimate/diff-coverage` are stuck for a while. They are not required, so I'm merging.

File: src/api/config/routes/webui_routes.rb (hunk 1 above)
Comment (saraycp) [reply src/api/config/routes/webui_routes.rb:229]: Oh, sorry! I didn't see your comment before merging. 
I'll remove it in another PR.

**Pull request merged**